spec/index.html

```diff
@@ -1493,3 +1503,221 @@ <h1>SharedArray ( ..._values_ )</h1>
+  </emu-clause>
+
+  <emu-clause id="sec-mutex-object">
+    <h1>The Mutex Constructor</h1>
+    <p>The Mutex constructor:</p>
+    <ul>
+      <li>is <dfn>%Mutex%</dfn>.</li>
+      <li>is the initial value of the *"Mutex"* property of the %Atomics% object.</li>
+      <li>creates and initializes a new Mutex when called as constructor.</li>
+      <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+    </ul>
+
+    <emu-clause id="sec-atomics.mutex">
+      <h1>Atomics.Mutex ( )</h1>
+      <p>This function performs the following steps when called:</p>
+      <emu-alg>
+        1. If NewTarget is *undefined*, throw a *TypeError* exception.
+        1. Let _createInitializer_ be a new Abstract Closure with parameters (_newMutex_) that captures nothing and performs the following steps when called:
+          1. Set _newMutex_.[[MutexWaiterList]] to a new WaiterList Record.
+          1. Set _newMutex_.[[IsLocked]] to *false*.
+          1. Return ~unused~.
+        1. Return SharedStructCreate(_createInitializer_, « [[MutexWaiterList]], [[IsLocked]] »).
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-the-mutex-constructor">
+    <h1>Properties of the Mutex Constructor</h1>
+    <p>The Mutex constructor:</p>
+    <ul>
+      <li>has a [[Prototype]] internal slot whose value is %Function.prototype%.</li>
+      <li>has the following properties:</li>
+    </ul>
+
+    <emu-note>
+      <p>Per-Realm prototypes, which is currently an open design question and not included in this draft, will give Mutexes a per-Realm prototype with built-in methods instead of static methods, and will also add support for the %Symbol.dispose%.</p>
+    </emu-note>
+
+    <emu-clause id="sec-atomics.mutex.lock">
+      <h1>Atomics.Mutex.lock ( _mutex_ [ , _unlockToken_ [ , _timeout_ ] ] )</h1>
+      <p>This function puts the surrounding agent in a wait queue and suspends it until the mutex is unlocked or until the wait times out, returning a String differentiating those cases.</p>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. If _mutex_ is not an Object, throw a *TypeError* exception.
+        1. If _mutex_ does not have a [[MutexWaiterList]] internal slot, throw a *TypeError* exception.
+        1. If _unlockToken_ not *undefined*, then
+          1. If _unlockToken_ is not an Object, throw a *TypeError* exception.
+          1. If _unlockToken_ does not have a [[LockedMutex]] internal slot, throw a *TypeError* exception.
+          1. If _unlockToken_.[[LockedMutex]] is not ~empty~, throw a *TypeError* exception.
+        1. If _timeout_ is *undefined*, set _timeout_ to *+∞*<sub>𝔽</sub>.
+        1. If _timeout_ is not a Number, throw a *TypeError* exception.
+        1. If _timeout_ is either *NaN* or *+∞*<sub>𝔽</sub>, let _t_ be +∞; else if _timeout_ is *-∞*<sub>𝔽</sub>, let _t_ be 0; else let _t_ be max(ℝ(_timeout_), 0).
```

Comment:
[NaN](https://github.com/tc39/how-we-work/blob/main/normative-conventions.md#number-taking-inputs-should-reject-nan) and negative values should be a RangeError IMO.

Reply:
This is aligned with how Atomics.wait works. I don't think we should diverge.

Reply:
Ah, wfm.